examples/batch/batch.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pgx "github.com/jackc/pgx/v5"
+	pgxpool "github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close()
+}
+
+func databaseSetup(db PgxIface) (err error) {
+
+	// Initialize a database object
+	tx, err := db.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("databaseSetup: %s", err)
+	}
+	// Finally, commit changes or rollback
+	defer func() {
+		switch err {
+		case nil:
+			err = tx.Commit(context.Background())
+		default:
+			_ = tx.Rollback(context.Background())
+		}
+	}()
+
+	// Create a new table 'ledger'
+	sql := `CREATE TABLE IF NOT EXISTS ledger (
+		id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+		description TEXT NOT NULL,
+		amount BIGINT NOT NULL);`
+
+	// Execute SQL commands
+	_, err = tx.Exec(context.Background(), sql)
+	if err != nil {
+		return fmt.Errorf("databaseSetup: %s", err)
+	}
+
+	return
+}
+
+func requestBatch(db PgxIface) (err error) {
+
+	// Initialize a database object
+	tx, err := db.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("requestBatch: %s", err)
+	}
+	// Finally, commit changes or rollback
+	defer func() {
+		switch err {
+		case nil:
+			err = tx.Commit(context.Background())
+		default:
+			_ = tx.Rollback(context.Background())
+		}
+	}()
+
+	// Create a Batch object
+	batch := &pgx.Batch{}
+
+	// Add SQL commands to queue
+	batch.Queue("insert into ledger(description, amount) values($1, $2)", "q1", 1)
+	batch.Queue("insert into ledger(description, amount) values($1, $2)", "q2", 2)
+	batch.Queue("insert into ledger(description, amount) values($1, $2)", "q3", 3)
+	batch.Queue("select id, description, amount from ledger order by id")
+	batch.Queue("select id, description, amount from ledger order by amount")
+	batch.Queue("select * from ledger where false")
+	batch.Queue("select sum(amount) from ledger")
```

Comment:
Thanks a lot for your work! 

Should we also add some logic for returned data using [BatchResults](https://pkg.go.dev/github.com/jackc/pgx/v5#BatchResults). At least for select statements. Otherwise people might not understand the idea behind batching.